### PR TITLE
add accuracy for each training batch

### PR DESCRIPTION
**Pull request by astorfi**

### CODE REVIEW

File: slim/train_image_classifier.py [hunk 1]
```diff
@@ -457,12 +457,21 @@ def clone_fn(batch_queue):
       # Specify the loss function #
       #############################
       if 'AuxLogits' in end_points:
-        tf.losses.softmax_cross_entropy(
-            logits=end_points['AuxLogits'], onehot_labels=labels,
-            label_smoothing=FLAGS.label_smoothing, weights=0.4, scope='aux_loss')
+          logits = end_points['AuxLogits']
```
Comment (nealwu) [inline slim/train_image_classifier.py]: This isn't correct. You changed the value of the variable `logits` for the operations below.

File: slim/train_image_classifier.py [hunk 2]
```diff
@@ -457,12 +457,21 @@ def clone_fn(batch_queue):
...
+          tf.losses.softmax_cross_entropy(
+              logits=logits['AuxLogits'], onehot_labels=labels,
```
Comment (nealwu) [inline slim/train_image_classifier.py]: This is also incorrect since you're calling `['AuxLogits']` twice.

File: slim/train_image_classifier.py [hunk 3]
```diff
@@ -457,12 +457,21 @@ def clone_fn(batch_queue):
...
       tf.losses.softmax_cross_entropy(
           logits=logits, onehot_labels=labels,
           label_smoothing=FLAGS.label_smoothing, weights=1.0)
+
+      #############################
+      ## Calculation of accuracy ##
+      #############################
+      accuracy = slim.metrics.accuracy(tf.to_int32(tf.argmax(logits, 1)),
+                                       tf.to_int32(tf.argmax(labels, 1)))
+      tf.add_to_collection('accuracy', accuracy)
```
Comment (nealwu) [inline slim/train_image_classifier.py]: In fact, I believe you only need this portion and not any of the code above this. You have everything you need to compute accuracy here and can output it after that.

File: slim/train_image_classifier.py (hunk 1 above)
Comment (astorfi) [inline slim/train_image_classifier.py]: The variable `logits` will only change if its above if condition is True which is part of the code.

Comment (astorfi) [inline slim/train_image_classifier.py]: However it needs another change

File: slim/train_image_classifier.py (hunk 2 above)
Comment (astorfi) [inline slim/train_image_classifier.py]: About this you are absolutely correct!

File: slim/train_image_classifier.py (hunk 3 above)
Comment (astorfi) [inline slim/train_image_classifier.py]: Sure ... However, that only determines the accuracy for a single clone not the average on all clones!
The main thing is the returning the averaged accuracy for all clones if we are dealing with mutli-gpu scenario!

File: slim/train_image_classifier.py (hunk 1 above)
Comment (astorfi) [inline slim/train_image_classifier.py]: The logits must be changed if the condition is True because we have to feed the new logits to calculated accuracy as it is needed to be fed for calculation of softmax cross-entropy loss.

**Comment by astorfi** [comment]
I made some changes. There is no need to change 'model_deploy.py' file now!
Only few lines have been changed in **slim/train_image_classifier.py** file.
Although GitHub shows lots of changes but it's only due to changes in line places or indentation most probably.

**Comment by astorfi** [comment]
Indentation mismatches have been corrected!

**Comment by astorfi** [comment]
Besides, flower fine tuning script for vgg_19 has been added.

**Comment by astorfi** [comment]
Any feedback?

File: slim/train_image_classifier.py [hunk 4]
```diff
@@ -457,12 +457,21 @@ def clone_fn(batch_queue):
...
-            label_smoothing=FLAGS.label_smoothing, weights=0.4, scope='aux_loss')
+            logits = end_points['AuxLogits']
+            tf.losses.softmax_cross_entropy(
+                logits=logits, onehot_labels=labels,
+                label_smoothing=FLAGS.label_smoothing, weights=0.4, scope='aux_loss')
       tf.losses.softmax_cross_entropy(
-          logits=logits, onehot_labels=labels,
-          label_smoothing=FLAGS.label_smoothing, weights=1.0)
+            logits=logits, onehot_labels=labels,
+            label_smoothing=FLAGS.label_smoothing, weights=1.0)
```
Comment (nealwu) [inline slim/train_image_classifier.py]: You still changed the behavior of this code when the above `if` is `True`. Are you suggesting that the code was incorrect previously?

Comment (nealwu) [inline slim/train_image_classifier.py]: Also please fix the indenting. You added six-space indenting here.

File: slim/train_image_classifier.py [hunk 5]
```diff
@@ -457,12 +457,21 @@ def clone_fn(batch_queue):
...
+
+      #############################
+      ## Calculation of accuracy ##
+      #############################
+      accuracy = slim.metrics.accuracy(tf.to_int32(tf.argmax(logits, 1)),
+                                         tf.to_int32(tf.argmax(labels, 1)))
```
Comment (nealwu) [inline slim/train_image_classifier.py]: I don't think the `tf.to_int32` is needed here. Also please fix the indentation so that the second line aligns properly with the first.

File: slim/train_image_classifier.py (hunk 4 above)
Comment (astorfi) [inline slim/train_image_classifier.py]: You are right ... Now it's fixed!

File: slim/train_image_classifier.py (hunk 5 above)
Comment (astorfi) [inline slim/train_image_classifier.py]: I tested that ... Looks like it's not necessary but I think eliminating  tf.to_int32 caused me some issue  ... For now I remove it though!

**Comment by astorfi** [comment]
All aforementioned issues been fixed!

**Comment by astorfi** [comment]
Is there any further action required?

**Comment by nealwu** [comment]
@sguada does this look appropriate to you?

**Comment by astorfi** [comment]
@nealwu  @sguada Any feedback?

**Comment by nealwu** [comment]
@sguada Let us know your feedback.

**Comment by astorfi** [comment]
@nealwu @sguada  Any feedback? Thanks

**Comment by sguada** [comment]
Please sync fix conflits.
